src/components/Cell.jsx

```diff
@@ -1,9 +1,51 @@
-
+//import { useLongPress } from '@use-gesture/react';
+import { useRef, useCallback } from 'react';
 import classNames from "classnames";
 import { Bomb, X, Flag } from "@phosphor-icons/react";
 
-function Cell({ cell, onClick, onContextMenu }) {
 
+export function useLongPress(callback = () => {}, threshold = 500) {
+  const timerRef = useRef();
+  const startTimeRef = useRef();
+
+  const start = useCallback((event) => {
+    event.preventDefault();
+    startTimeRef.current = Date.now();
+    
+    timerRef.current = setTimeout(() => {
+      timerRef.current = null;// Does not sound like making sense.
+      callback(event);//Does do it twice? Or halfway?
+    }, threshold);
+  }, [callback, threshold]);
+  
+  const end = useCallback(() => {
+    const duration = Date.now() - startTimeRef.current;
+
+    if (duration < threshold) {
+      clearTimeout(timerRef.current);
+    }
+  }, [threshold]);
+
+  return () => ({
+    //onPointerDown: start,
+    //onPointerUp: leave,
+    onTouchStart: start,
+    onTouchEnd: end,
+    onTouchCancel: end,
+  });
+}
+
+function Cell({ cell, onClick, onContextMenu }) {
+  const bind = useLongPress(
+    (event) => {
+    // Handle long press as a right-click or flag gesture
+      onContextMenu(event);
+    },
+    {
+      threshold: 500, // Long press threshold in milliseconds
+    }
```

Comment:
seems like you're sending an object here, but your hook expects to get the value instead, like this:

```suggestion
    500
```